api-service/src/controllers/CreateQueryTemplate/QueryTemplateValidator.ts

```diff
@@ -0,0 +1,57 @@
+import * as _ from "lodash";
+import { config } from "../../configs/Config";
+
+export const validateTemplate = async (req: Request) => {
+    const type: any = _.get(req, "request.query_type");
+    const query = _.get(req, 'request');
+    const templateData: string = JSON.stringify(query);
+    const validTemplate = isValidTemplate(templateData, type);
+    return { validTemplate };
+}
+
+const isValidTemplate = (templateData: string, type: string) => {
+    let validTemplate = false;
+    const requiredVars = requiredVariablesExist(config?.template_config?.template_required_variables, getTemplateVariables(templateData));
+    if (!requiredVars) return validTemplate;
+    if (type === "json") {
+        let data = _.cloneDeep(stringifyVars(templateData, type));
+        getTemplateVariables(data).map((variable: string) => {
+            const varRegex = new RegExp(`"{{${variable}}}"`, 'ig');
+            data = data.replace(varRegex, `""`);
+            data = data.replace(/"""/g, `"`);
+        });
+        try {
+            JSON.parse(data);
+            validTemplate = true;
+        } catch (err: any) { validTemplate = false; }
```

Comment:
Still this error is not logged

Reply:
removed try catch while parsing, because we are validation json in schema only